test/powershell/Modules/Microsoft.PowerShell.Utility/Get-FileHash.Tests.ps1

```diff
@@ -3,15 +3,23 @@
 Describe "Get-FileHash" -Tags "CI" {
 
     BeforeAll {
-        $testDocument = Join-Path -Path $PSScriptRoot -ChildPath assets testablescript.ps1
-        Write-Host $testDocument
+        $testDocument = Join-Path -Path $TestDrive -ChildPath "hashable.txt"
+        $utf8NoBOM = [System.Text.UTF8Encoding]::new($false)
+        [System.IO.File]::WriteAllText($testDocument, "Get-Module`n", $utf8NoBOM)
```

Comment:
Excellent questions!

The default Git configuration on Windows machines means that LF is checked out as CRLF -- which added a CR byte to `testablescript.ps1` and broke the test. Also, opening testablescript.ps1 in text editors caused the BOM to get deleted in some cases.

I would have been happy with leaving the BOM, or even just writing some arbitrary bytes to a file, but we decided that if .NET Core is trying to get away from using BOMs in UTF-8, we should encourage that everywhere. Also, it makes the tests easier to verify against other hashsum calculators in BOM-less UTF-8 environments, which are arguably more prevalent and more likely to have an independent implementation from ours.

Finally, the WriteAllText we use is [`[System.IO.File]::WriteAllText(String path, String text, Encoding fileEncoding)`](https://msdn.microsoft.com/en-us/library/ms143376(v=vs.110).aspx), rather than [`[Microsoft.VisualBasic.FileIO.FileSystem]::WriteAllText(String file, String text, bool append)`](https://msdn.microsoft.com/en-us/library/ms128035%28v=vs.110%29.aspx). Which I also got caught up by a couple of times.

Reply:
The hash should be based on the bytes of the file and the file encoding including BOM is opaque to the cmdlet, so don't think there's value in having a test with BOM.